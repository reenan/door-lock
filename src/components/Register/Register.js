import React from 'react'
import './Register.scss'

import { Icon, Modal, Input, Button } from '../'

export default ({
  registerModalIsOpen,
  storeName,
  openRegisterModal,
  closeRegisterModal,
  handleStoreNameChange,
  registerStore,
}) => (
  <div className='register'>
    <div className='hero' />

    <div className='content'>
      <div className='top'>
        <p className='big'>Hello,</p>
        <p className='medium'>Welcome to the <span>Door Lock</span> application.</p>
        <p className='small'>
        Your store's security matters, that is why we are here to help you ensure
        that no one without permissions will be opening your doors.
        </p>

      </div>

      <div className='bottom'>
        <p className='call'>It seems like you have not defined your store yet, how about starting now?</p>

        <div className='register-button' onClick={openRegisterModal}>
          <Icon size={22} icon='lock-open' />
          <Icon size={22} icon='lock' />
          <span>Protect my store</span>
        </div>
      </div>
    </div>

    <Modal open={registerModalIsOpen} close={closeRegisterModal}>
      <p className='medium'>Awesome! You just did the first step.</p>

      <Input value={storeName} onChange={handleStoreNameChange}
        label="What is your store's name?" required />

        <Button icon='angle-right' text='Register' onClick={registerStore}
          className={storeName.length > 0 ? '' : 'disabled'} />

    </Modal>
  </div>
)
