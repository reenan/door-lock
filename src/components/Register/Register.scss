@import '../../resources/_mixins.scss';

.register {
  width: 80%;
  margin: auto;

  .hero {
    background-image: linear-gradient(90deg, #263869 20%, #4b6cb7);
    position: absolute;
    left: 0;
    right: 0;
    top: 0;
    height: 200px;
    padding: 80px 80px 100px 80px;

    &:after {
      border-width: 90px 0 0 100vw;
      content: "";
      position: absolute;
      bottom: 0;
      left: 0;
      width: 0;
      height: 0;
      border-style: solid;
      border-width: 50px 0 0 100vw;
      border-color: transparent transparent transparent #0e1938;
      z-index: 2;
    }
  }

  .content {
    position: relative;
    z-index: 2;

    .top {
      margin: 100px 0 200px 0;

      p {
        opacity: 0;
        transform: translateY(20px);
        animation: fade-in-up 0.6s;
        animation-fill-mode: forwards;

        &.medium {
          animation-delay: 0.2s;
        }

        &.small {
          animation-delay: 0.4s;
        }
      }
    }

    .bottom {
      opacity: 0;
      transform: translateY(20px);
      animation: fade-in-up 0.6s 0.6s;
      animation-fill-mode: forwards;

      .register-button {
        @include transition(background-color 600ms);
        display: inline-block;
        border: 1px solid #25cb68;
        border-radius: 45px;
        padding: 30px 45px;
        margin-top: 30px;
        cursor: pointer;
        position: relative;

        span {
          @include vertical-align(middle);
          text-transform: uppercase;

          &.icon {
            @include transition(opacity 400ms);
            position: absolute;
            color: #25cb68;

            &.lock {
              opacity: 0;
            }
          }

          &:not(.icon) {
            margin-left: 40px;
          }
        }

        &:hover {
         background-color: rgba(#4b6cb7, 0.5);

         span.icon {
           &.lock {
             opacity: 1;
           }

           &.lock-open {
             opacity: 0;
           }
         }

        }
      }
    }
  }


  .modal {
    input {
      width: 400px;
      margin: 45px 0 0 0;
    }

    .button {
      float: right;
    }
  }

  p {
    margin-bottom: 15px;

    &.big {
      color: #25cb68;
      font-size: 50px;
      font-weight: bold;
    }

    &.medium {
      font-size: 22px;
    }

    &.small {
      font-size: 16px;
    }


    &.call {
      font-size: 22px;
      font-style: italic;
    }

    > span {
      font-weight: bold;
      font-style: italic;
    }
  }
}
