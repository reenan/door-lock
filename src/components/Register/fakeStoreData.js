import uuid4 from 'uuid4'

const managerID = uuid4()
const buyerID = uuid4()
const cashierID = uuid4()
const clientID = uuid4()

const roles = {
  [managerID]: { name: 'Manager', permissions: { frontDoor: true, storageRoom: true } },
  [buyerID]: { name: 'Buyer', permissions: { frontDoor: false, storageRoom: true } },
  [cashierID]: { name: 'Cashier', permissions: { frontDoor: true, storageRoom: false } },
  [clientID]: { name: 'Client', permissions: { frontDoor: false, storageRoom: false } },
}

const JaneID = uuid4()
const JohnID = uuid4()
const RichardID = uuid4()
const AnnaID = uuid4()

const employees = {
  [JaneID]: { name: 'Jane', role: managerID },
  [JohnID]: { name: 'John', role: buyerID },
  [RichardID]: { name: 'Richard', role: cashierID },
  [AnnaID]: { name: 'Anna', role: clientID },
}

export { roles, employees }