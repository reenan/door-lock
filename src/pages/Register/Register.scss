.register {
  height: 100%;
  
  .ui.vertical.segment {
    padding: 3em 0;
  }
  
  .ui.stackable.steps.step-group {
    margin: 30px 0;
    
    .step {
      max-width: 300px;
      padding-bottom: 2em;
      
      > .icon {
        margin: 15px 0;
      }
    }
  }
  
  .ui.inverted.header {
    &.call {
      font-size: 1.6em;
      font-style: italic;
      margin-top: 0.5em;
      margin-bottom: 1.5em;
    }
  }
}
