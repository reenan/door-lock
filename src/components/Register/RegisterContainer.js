import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Redirect } from 'react-router-dom'

import { registerStore } from '../../actions'
import Register from './Register'

class RegisterContainer extends Component {

  constructor(props) {
    super(props)

    this.state = {
      registerModalIsOpen: false,
      storeName: '',
    }
  }

  registerStore = () => {
    this.props.dispatch(registerStore())
  }

  openRegisterModal = () => {
    this.setState({
      registerModalIsOpen: true
    })
  }

  closeRegisterModal = () => {
    this.setState({
      registerModalIsOpen: false
    })
  }

  handleStoreNameChange = (e) => {
    this.setState({
      storeName: e.target.value
    })
  }

  render() {
    const { registerModalIsOpen, storeName } = this.state
    const { store } = this.props

    return (
      store.name ?
        <Redirect to='/store' /> :
        <Register
          registerModalIsOpen={registerModalIsOpen}
          storeName={storeName}
          openRegisterModal={this.openRegisterModal}
          closeRegisterModal={this.closeRegisterModal}
          handleStoreNameChange={this.handleStoreNameChange}
        />

    )
  }
}

RegisterContainer.propTypes = {
  dispatch: PropTypes.func.isRequired,
  store: PropTypes.object.isRequired
}

const stateToProps = ({ store }) => ({
  store
})

export default connect(stateToProps)(RegisterContainer)
