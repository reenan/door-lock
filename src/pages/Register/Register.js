import React from 'react'
import {
  Button,
  Container,
  Header,
  Segment,
  Step,
  Icon,
  Modal,
  Input,
  Loader,
  Dimmer,
} from 'semantic-ui-react'

import './Register.scss'

export default ({
  isOpenRegisterModal,
  storeName,
  loading,
  openRegisterModal,
  closeRegisterModal,
  handleStoreNameChange,
  registerStore,
}) => (
  <div className='register'>
    <Segment inverted textAlign='center' vertical>
      <Container text>
        <Header as='h1' size='huge' inverted>
          Welcome to The Cloud Lock
        </Header>

        <Header as='h4' inverted className='call'>
          Protect your store's doors with assured security and open them with one click!
        </Header>

        <Header.Subheader>
          Your store's security matters, that is why we are here to help you ensure that no one
          without permissions will be opening your doors.
        </Header.Subheader>
      </Container>

      <Step.Group className='step-group' stackable='tablet'>
        <Step>
          <Icon name='star' />
          <Step.Content>
            <Step.Title>Register your store</Step.Title>
            <Step.Description>
              Simply enter your store's name and we handle the rest.
            </Step.Description>
          </Step.Content>
        </Step>
        <Step>
          <Icon name='id badge' />
          <Step.Content>
            <Step.Title>Customize</Step.Title>
            <Step.Description>
              Tell us what are your doors, your employees, their roles and authorisations.
            </Step.Description>
          </Step.Content>
        </Step>
        <Step>
          <Icon name='lock' />
          <Step.Content>
            <Step.Title>That's it!</Step.Title>
            <Step.Description>
              Your store is secured and your employees can open doors with one click.
            </Step.Description>
          </Step.Content>
        </Step>
      </Step.Group>

       <Container text>
        <Header as='h4' inverted className='call'>
          It looks like you've not registered your store yet, how about starting it now?
        </Header>

        <Button animated size='huge' onClick={openRegisterModal}>
          <Button.Content visible>Protect my store</Button.Content>
          <Button.Content hidden>
            <Icon name='lock' />
          </Button.Content>
        </Button>
      </Container>

    </Segment>

    <Modal size='tiny' open={isOpenRegisterModal} onClose={closeRegisterModal}>

      <Modal.Header>Register your store</Modal.Header>
      <Modal.Content>
        <p>Awesome! You just did the first step.</p>
        <p>Now, please, could you tell us what is your store's name?</p>

        <Input placeholder='Store Name'
          value={storeName} onChange={handleStoreNameChange} />

        <Dimmer active={loading} >
          <Loader/>
        </Dimmer>

      </Modal.Content>
      <Modal.Actions>
        <Button positive disabled={storeName.length === 0}  icon='angle right'
          labelPosition='right' content='Register' onClick={registerStore} />
      </Modal.Actions>

    </Modal>
  </div>
)
