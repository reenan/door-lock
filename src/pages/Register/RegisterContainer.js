import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Redirect } from 'react-router-dom'

import { registerStore } from '../../actions'
import Register from './Register'

class RegisterContainer extends Component {

  constructor(props) {
    super(props)

    this.state = {
      isOpenRegisterModal: false,
      storeName: '',
      loading: false,
    }
  }

  registerStore = () => {
    const { storeName } = this.state

    this.props.dispatch(registerStore(storeName)).then(() => {
      this.props.history.push('/store')
    })
  }

  openRegisterModal = () => {
    this.setState({
      isOpenRegisterModal: true
    })
  }

  closeRegisterModal = () => {
    this.setState({
      isOpenRegisterModal: false
    })
  }

  handleStoreNameChange = (e) => {
    this.setState({
      storeName: e.target.value
    })
  }

  render() {
    const { isOpenRegisterModal, storeName } = this.state
    const { store, isFetching } = this.props

    return (
      store.name ?
        <Redirect to='/store' /> :
        <Register
          loading={isFetching}
          isOpenRegisterModal={isOpenRegisterModal}
          storeName={storeName}
          openRegisterModal={this.openRegisterModal}
          closeRegisterModal={this.closeRegisterModal}
          handleStoreNameChange={this.handleStoreNameChange}
          registerStore={this.registerStore}
        />
    )
  }
}

RegisterContainer.propTypes = {
  dispatch: PropTypes.func.isRequired,
  store: PropTypes.object.isRequired
}

const stateToProps = ({ store, isFetching }) => ({
  store,
  isFetching,
})

export default connect(stateToProps)(RegisterContainer)
