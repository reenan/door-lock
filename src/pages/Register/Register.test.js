import React from 'react'
import Register from './Register'

import EnzymeToJson from 'enzyme-to-json'
import { mount } from 'enzyme'

it('renders without crashing', () => {
  const component = mount(
    <Register
      registerModalIsOpen={false}
      storeName=''
      openRegisterModal={() => {}}
      closeRegisterModal={() => {}}
      handleStoreNameChange={() => {}}
    />
  )

  expect(EnzymeToJson(component)).toMatchSnapshot()
})

it('renders without crashing, modal open', () => {
  const component = mount(
    <Register
      registerModalIsOpen={true}
      storeName='Mock'
      openRegisterModal={() => {}}
      closeRegisterModal={() => {}}
      handleStoreNameChange={() => {}}
    />
  )

  expect(EnzymeToJson(component)).toMatchSnapshot()
})
