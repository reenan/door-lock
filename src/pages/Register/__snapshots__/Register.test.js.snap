// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<_default
  closeRegisterModal={[Function]}
  handleStoreNameChange={[Function]}
  openRegisterModal={[Function]}
  registerModalIsOpen={false}
  storeName=""
>
  <div
    className="register"
  >
    <Segment
      inverted={true}
      textAlign="center"
      vertical={true}
    >
      <div
        className="ui inverted vertical center aligned segment"
      >
        <Container
          text={true}
        >
          <div
            className="ui text container"
          >
            <Header
              as="h1"
              inverted={true}
              size="huge"
            >
              <h1
                className="ui huge inverted header"
              >
                Welcome to The Cloud Lock
              </h1>
            </Header>
            <Header
              as="h4"
              className="call"
              inverted={true}
            >
              <h4
                className="ui inverted header call"
              >
                Protect your store's doors with assured security and open them with one click!
              </h4>
            </Header>
            <HeaderSubheader>
              <div
                className="sub header"
              >
                Your store's security matters, that is why we are here to help you ensure that no one without permissions will be opening your doors.
              </div>
            </HeaderSubheader>
          </div>
        </Container>
        <StepGroup
          className="step-group"
          stackable="tablet"
        >
          <div
            className="ui tablet stackable steps step-group"
          >
            <Step>
              <div
                className="step"
                onClick={[Function]}
              >
                <Icon
                  as="i"
                  name="star"
                >
                  <i
                    aria-hidden="true"
                    className="star icon"
                    onClick={[Function]}
                  />
                </Icon>
                <StepContent>
                  <div
                    className="content"
                  >
                    <StepTitle>
                      <div
                        className="title"
                      >
                        Register your store
                      </div>
                    </StepTitle>
                    <StepDescription>
                      <div
                        className="description"
                      >
                        Simply enter your store's name and we handle the rest.
                      </div>
                    </StepDescription>
                  </div>
                </StepContent>
              </div>
            </Step>
            <Step>
              <div
                className="step"
                onClick={[Function]}
              >
                <Icon
                  as="i"
                  name="id badge"
                >
                  <i
                    aria-hidden="true"
                    className="id badge icon"
                    onClick={[Function]}
                  />
                </Icon>
                <StepContent>
                  <div
                    className="content"
                  >
                    <StepTitle>
                      <div
                        className="title"
                      >
                        Customize
                      </div>
                    </StepTitle>
                    <StepDescription>
                      <div
                        className="description"
                      >
                        Tell us what are your doors, your employees, their roles and authorisations.
                      </div>
                    </StepDescription>
                  </div>
                </StepContent>
              </div>
            </Step>
            <Step>
              <div
                className="step"
                onClick={[Function]}
              >
                <Icon
                  as="i"
                  name="lock"
                >
                  <i
                    aria-hidden="true"
                    className="lock icon"
                    onClick={[Function]}
                  />
                </Icon>
                <StepContent>
                  <div
                    className="content"
                  >
                    <StepTitle>
                      <div
                        className="title"
                      >
                        That's it!
                      </div>
                    </StepTitle>
                    <StepDescription>
                      <div
                        className="description"
                      >
                        Your store is secured and your employees can open doors with one click.
                      </div>
                    </StepDescription>
                  </div>
                </StepContent>
              </div>
            </Step>
          </div>
        </StepGroup>
        <Container
          text={true}
        >
          <div
            className="ui text container"
          >
            <Header
              as="h4"
              className="call"
              inverted={true}
            >
              <h4
                className="ui inverted header call"
              >
                It looks like you've not registered your store yet, how about starting it now?
              </h4>
            </Header>
            <Button
              animated={true}
              as="button"
              onClick={[Function]}
              size="huge"
            >
              <Ref
                innerRef={
                  Object {
                    "current": <button
                      class="ui huge animated button"
                    >
                      <div
                        class="visible content"
                      >
                        Protect my store
                      </div>
                      <div
                        class="hidden content"
                      >
                        <i
                          aria-hidden="true"
                          class="lock icon"
                        />
                      </div>
                    </button>,
                  }
                }
              >
                <RefFindNode
                  innerRef={
                    Object {
                      "current": <button
                        class="ui huge animated button"
                      >
                        <div
                          class="visible content"
                        >
                          Protect my store
                        </div>
                        <div
                          class="hidden content"
                        >
                          <i
                            aria-hidden="true"
                            class="lock icon"
                          />
                        </div>
                      </button>,
                    }
                  }
                >
                  <button
                    className="ui huge animated button"
                    onClick={[Function]}
                  >
                    <ButtonContent
                      visible={true}
                    >
                      <div
                        className="visible content"
                      >
                        Protect my store
                      </div>
                    </ButtonContent>
                    <ButtonContent
                      hidden={true}
                    >
                      <div
                        className="hidden content"
                      >
                        <Icon
                          as="i"
                          name="lock"
                        >
                          <i
                            aria-hidden="true"
                            className="lock icon"
                            onClick={[Function]}
                          />
                        </Icon>
                      </div>
                    </ButtonContent>
                  </button>
                </RefFindNode>
              </Ref>
            </Button>
          </div>
        </Container>
      </div>
    </Segment>
    <Modal
      centered={true}
      className="register-modal"
      closeOnDimmerClick={true}
      closeOnDocumentClick={false}
      dimmer={true}
      eventPool="Modal"
      onClose={[Function]}
      size="tiny"
    >
      <Portal
        closeOnDocumentClick={false}
        closeOnEscape={true}
        eventPool="Modal"
        mountNode={<body />}
        onClose={[Function]}
        onMount={[Function]}
        onOpen={[Function]}
        onUnmount={[Function]}
        openOnTriggerClick={true}
      />
    </Modal>
  </div>
</_default>
`;

exports[`renders without crashing, modal open 1`] = `
<_default
  closeRegisterModal={[Function]}
  handleStoreNameChange={[Function]}
  openRegisterModal={[Function]}
  registerModalIsOpen={true}
  storeName="Mock"
>
  <div
    className="register"
  >
    <Segment
      inverted={true}
      textAlign="center"
      vertical={true}
    >
      <div
        className="ui inverted vertical center aligned segment"
      >
        <Container
          text={true}
        >
          <div
            className="ui text container"
          >
            <Header
              as="h1"
              inverted={true}
              size="huge"
            >
              <h1
                className="ui huge inverted header"
              >
                Welcome to The Cloud Lock
              </h1>
            </Header>
            <Header
              as="h4"
              className="call"
              inverted={true}
            >
              <h4
                className="ui inverted header call"
              >
                Protect your store's doors with assured security and open them with one click!
              </h4>
            </Header>
            <HeaderSubheader>
              <div
                className="sub header"
              >
                Your store's security matters, that is why we are here to help you ensure that no one without permissions will be opening your doors.
              </div>
            </HeaderSubheader>
          </div>
        </Container>
        <StepGroup
          className="step-group"
          stackable="tablet"
        >
          <div
            className="ui tablet stackable steps step-group"
          >
            <Step>
              <div
                className="step"
                onClick={[Function]}
              >
                <Icon
                  as="i"
                  name="star"
                >
                  <i
                    aria-hidden="true"
                    className="star icon"
                    onClick={[Function]}
                  />
                </Icon>
                <StepContent>
                  <div
                    className="content"
                  >
                    <StepTitle>
                      <div
                        className="title"
                      >
                        Register your store
                      </div>
                    </StepTitle>
                    <StepDescription>
                      <div
                        className="description"
                      >
                        Simply enter your store's name and we handle the rest.
                      </div>
                    </StepDescription>
                  </div>
                </StepContent>
              </div>
            </Step>
            <Step>
              <div
                className="step"
                onClick={[Function]}
              >
                <Icon
                  as="i"
                  name="id badge"
                >
                  <i
                    aria-hidden="true"
                    className="id badge icon"
                    onClick={[Function]}
                  />
                </Icon>
                <StepContent>
                  <div
                    className="content"
                  >
                    <StepTitle>
                      <div
                        className="title"
                      >
                        Customize
                      </div>
                    </StepTitle>
                    <StepDescription>
                      <div
                        className="description"
                      >
                        Tell us what are your doors, your employees, their roles and authorisations.
                      </div>
                    </StepDescription>
                  </div>
                </StepContent>
              </div>
            </Step>
            <Step>
              <div
                className="step"
                onClick={[Function]}
              >
                <Icon
                  as="i"
                  name="lock"
                >
                  <i
                    aria-hidden="true"
                    className="lock icon"
                    onClick={[Function]}
                  />
                </Icon>
                <StepContent>
                  <div
                    className="content"
                  >
                    <StepTitle>
                      <div
                        className="title"
                      >
                        That's it!
                      </div>
                    </StepTitle>
                    <StepDescription>
                      <div
                        className="description"
                      >
                        Your store is secured and your employees can open doors with one click.
                      </div>
                    </StepDescription>
                  </div>
                </StepContent>
              </div>
            </Step>
          </div>
        </StepGroup>
        <Container
          text={true}
        >
          <div
            className="ui text container"
          >
            <Header
              as="h4"
              className="call"
              inverted={true}
            >
              <h4
                className="ui inverted header call"
              >
                It looks like you've not registered your store yet, how about starting it now?
              </h4>
            </Header>
            <Button
              animated={true}
              as="button"
              onClick={[Function]}
              size="huge"
            >
              <Ref
                innerRef={
                  Object {
                    "current": <button
                      class="ui huge animated button"
                    >
                      <div
                        class="visible content"
                      >
                        Protect my store
                      </div>
                      <div
                        class="hidden content"
                      >
                        <i
                          aria-hidden="true"
                          class="lock icon"
                        />
                      </div>
                    </button>,
                  }
                }
              >
                <RefFindNode
                  innerRef={
                    Object {
                      "current": <button
                        class="ui huge animated button"
                      >
                        <div
                          class="visible content"
                        >
                          Protect my store
                        </div>
                        <div
                          class="hidden content"
                        >
                          <i
                            aria-hidden="true"
                            class="lock icon"
                          />
                        </div>
                      </button>,
                    }
                  }
                >
                  <button
                    className="ui huge animated button"
                    onClick={[Function]}
                  >
                    <ButtonContent
                      visible={true}
                    >
                      <div
                        className="visible content"
                      >
                        Protect my store
                      </div>
                    </ButtonContent>
                    <ButtonContent
                      hidden={true}
                    >
                      <div
                        className="hidden content"
                      >
                        <Icon
                          as="i"
                          name="lock"
                        >
                          <i
                            aria-hidden="true"
                            className="lock icon"
                            onClick={[Function]}
                          />
                        </Icon>
                      </div>
                    </ButtonContent>
                  </button>
                </RefFindNode>
              </Ref>
            </Button>
          </div>
        </Container>
      </div>
    </Segment>
    <Modal
      centered={true}
      className="register-modal"
      closeOnDimmerClick={true}
      closeOnDocumentClick={false}
      dimmer={true}
      eventPool="Modal"
      onClose={[Function]}
      size="tiny"
    >
      <Portal
        closeOnDocumentClick={false}
        closeOnEscape={true}
        eventPool="Modal"
        mountNode={<body />}
        onClose={[Function]}
        onMount={[Function]}
        onOpen={[Function]}
        onUnmount={[Function]}
        openOnTriggerClick={true}
      />
    </Modal>
  </div>
</_default>
`;
